doc/source/whatsnew/v0.24.0.rst

```diff
@@ -731,6 +731,38 @@ Bug in :func:`DataFrame.to_dict` raises ``ValueError`` when used with
 
     df.to_dict(orient='index')
 
+.. _whatsnew_0240.api_breaking.incompatible_index_unions
+
+Incompatible Index Type Unions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When performing an :func:`Index.union` operation, between objects of incompatible dtypes, the result will be a base :class:`Index` of dtype `object`. 
+This behavior holds true for unions between :class:`Index` objects that previously would have been prohibited.
+The dtype of empty :class:`Index` objects will now be evaluated before performing the union as well, instead of simply returning the other :class:`Index` object.
+
+
+Previous Behavior:
+
+.. code-block:: ipython
+
+    
+    In [1]: pd.period_range('19910905', periods=2).union(pd.Int64Index([1, 2, 3]))
+    ...
+    ValueError: can only call with other PeriodIndex-ed objects
+
+    In [2]: pd.Index([]).astype(str).union(pd.Index([1, 2, 3]))
+    Out[2]: Int64Index([1, 2, 3], dtype='int64')
+
+Current Behavior:
+
+.. code-block:: ipython
```

Comment:
use an ipython block so these are evaluated

Reply:
This should be `.. ipython:: python`

and just have the input. It'll be run as part of the doc build.

Reply:
fixed.